### PR TITLE
Split `test_parse`

### PR DESCRIPTION
**Pull request by pitdicker**
Combined with the extra tests on the main branch this test got so large it did not even compile on all targets (WASM).

### CODE REVIEW

File: src/format/parse.rs [hunk 1]
```diff
@@ -564,28 +564,26 @@ mod tests {
     use crate::format::*;
     use crate::{DateTime, FixedOffset, TimeZone, Timelike, Utc};
 
-    #[test]
-    fn test_parse() {
-        use crate::format::InternalInternal::*;
-        use crate::format::Item::{Literal, Space};
-        use crate::format::Numeric::*;
+    fn parse_all(s: &str, items: &[Item]) -> ParseResult<Parsed> {
```
Comment (djc) [inline src/format/parse.rs:567]: Style nit: helper below the tests, please.

File: src/format/parse.rs [hunk 2]
```diff
@@ -564,28 +564,26 @@ mod tests {
...
+        parse(&mut parsed, s, items.iter())?;
+        Ok(parsed)
+    }
 
-        fn parse_all(s: &str, items: &[Item]) -> ParseResult<Parsed> {
-            let mut parsed = Parsed::new();
-            parse(&mut parsed, s, items.iter())?;
-            Ok(parsed)
-        }
+    macro_rules! check {
```
Comment (djc) [inline src/format/parse.rs:573]: Unrelated to this particular change, but since you're cleaning this area up: does this need to be a macro?

**Review by djc — APPROVED** [review]
Nice!

File: src/format/parse.rs (hunk 1 above)
Comment (pitdicker) [reply src/format/parse.rs:567]: I'll learn at some point :smile:

File: src/format/parse.rs (hunk 2 above)
Comment (pitdicker) [reply src/format/parse.rs]: You thought: let's ask something easy? :laughing:
I still use a macro to created the `Parsed` structs to compare against, but the test now uses more regular rust code.

Comment (djc) [reply src/format/parse.rs]: I didn't ask you to change anything about it. 😋

**Pull request merged**